Recipe for Tunisian_MSA corpus.

johnjosephmorgan opened the pull request:
A recipe to build an ASR system with the Tunisian_MSA corpus of Arabic.

danpovey commented:
@jjm, there are some conflicts here that need to be resolved in git (i.e. you need to merge with master and resolve the conflicts).  Also, you seem to have copied the files from local/grammar/ over from librispeech to heroico-- please remove those.  Actually I see a large number of changes in the heroico recipe which I think are not relevant here-- if you want to make changes to that, could you please do it as a separate PR?  One possibility is to have someone here work on cleaning this up, if you don't have time.

entn-at commented on egs/tunisian_msa/s5/local/buckwalter2utf8.pl:
```diff
@@ -0,0 +1,11 @@
+use warnings;
+use Carp;
+
+use Encode::Arabic::Buckwalter;         # imports just like 'use Encode' would, plus more
```
On Ubuntu, package "libencode-arabic-perl" needs to be installed (https://packages.ubuntu.com/bionic/libencode-arabic-perl). Perhaps this requirement can be added to the README file (just like some recipes require ffmpeg to be installed).

entn-at commented on egs/tunisian_msa/s5/run.sh:
```diff
@@ -0,0 +1,192 @@
+if [ $stage -le 9 ]; then
+  # extract acoustic features
+  for fld in devtest train test; do
+    steps/make_plp_pitch.sh data/$fld exp/make_plp_pitch/$fld plp_pitch
```
I was just wondering, did PLP+pitch work better than MFCCs?

johnjosephmorgan replied:
I am using the Tunisian MSA corpus in recipes derived from the babel_multilang recipes which use plp_pitch.
Should I use mfcc instead?

entn-at replied:
I see, thanks! I was just wondering, since f0 (pitch) is more commonly used when working with tonal languages, which makes a lot of sense in the multilang setup.

entn-at commented on egs/tunisian_msa/s5/local/chain/tuning/run_tdnn_1a.sh:
```diff
@@ -0,0 +1,282 @@
+
+# The rest are configs specific to this script.  Most of the parameters
+# are just hardcoded at this level, in the commands below.
+affix=1b   # affix for the TDNN directory name
```
This should be `affix=1a`

johnjosephmorgan commented:
Sorry about that . I'll work on cleaning up.
J

On 9/19/18, Daniel Povey <notifications@github.com> wrote:
> @jjm, there are some conflicts here that need to be resolved in git (i.e.
> you need to merge with master and resolve the conflicts).  Also, you seem to
> have copied the files from local/grammar/ over from librispeech to heroico--
> please remove those.  Actually I see a large number of changes in the
> heroico recipe which I think are not relevant here-- if you want to make
> changes to that, could you please do it as a separate PR?  One possibility
> is to have someone here work on cleaning this up, if you don't have time.
>
> --
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub:
> https://github.com/kaldi-asr/kaldi/pull/2722#issuecomment-422870148

danpovey commented:
mfcc would be the normal default, yes.  Personally I'd recommend that you
use mini_librispeech as the template unless there is a good reason.

On Wed, Sep 19, 2018 at 1:54 PM John J Morgan <notifications@github.com>
wrote:

> *@johnjosephmorgan* commented on this pull request.
> ------------------------------
>
> In egs/tunisian_msa/s5/run.sh
> <https://github.com/kaldi-asr/kaldi/pull/2722#discussion_r218903011>:
>
> > +if [ $stage -le 7 ]; then
> +  echo "lm training."
> +  local/prepare_lm.sh  $tmpdir/subs/lm/in_vocabulary.txt
> +fi
> +
> +if [ $stage -le 8 ]; then
> +  echo "Making grammar fst."
> +  utils/format_lm.sh \
> +    data/lang data/local/lm/trigram.arpa.gz data/local/dict/lexicon.txt \
> +    data/lang_test
> +fi
> +
> +if [ $stage -le 9 ]; then
> +  # extract acoustic features
> +  for fld in devtest train test; do
> +    steps/make_plp_pitch.sh data/$fld exp/make_plp_pitch/$fld plp_pitch
>
> I am using the Tunisian MSA corpus in recipes derived from the
> babel_multilang recipes which use plp_pitch.
> Should I use mfcc instead?
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/2722#discussion_r218903011>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ADJVu-g0-r5qLUOjFsdb2i3NAaaIglbnks5ucoTLgaJpZM4Wwdhl>
> .
>

jjm commented:
Would you mind not tagging me on your PRs, I’m not @johnjosephmorgan 😉

johnjosephmorgan commented:
It looks like I was not able to resolve the conflicts. I accept help :)

danpovey commented:
OK @xiaohui-zhang  and @huangruizhe  will look at it.

entn-at commented:
The conflicts seem to be due to changes to files in the heroico recipe that don't seem to be related to the Tunisian MSA recipe.

xiaohui-zhang commented:
done in #2725

xiaohui-zhang commented on egs/tunisian_msa/s5/local/qcri_buckwalter2utf8.sh:
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+cut -d " " -f 1 qcri.txt > qcri_words_buckwalter.txt
```
this should probably be 
 cat qcri.txt | tail -n +4 | cut -d " " -f 1  > qcri_words_buckwalter.txt
 cat qcri.txt | tail -n +4 | cut -d " " -f 2 > qcri_prons.txt
Otherwise lines like "# Copyright" will be included.

johnjosephmorgan commented:
I was already doing this in the download script:
egs/Tunisian_msa/s5/local/qcri_lexicon_download.sh
I wasn't overwriting the unzipped text file with the header removed,
so I fixed that.
J

On 9/20/18, Xiaohui Zhang <notifications@github.com> wrote:
> xiaohui-zhang commented on this pull request.
>
>
>
>> @@ -0,0 +1,10 @@
> +#!/bin/bash
> +
> +cut -d " " -f 1 qcri.txt > qcri_words_buckwalter.txt
>
> this should probably be
>  cat qcri.txt | tail -n +4 | cut -d " " -f 1  > qcri_words_buckwalter.txt
>  cat qcri.txt | tail -n +4 | cut -d " " -f 2 > qcri_prons.txt
> Otherwise lines like "# Copyright" will be included.
>
> --
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub:
> https://github.com/kaldi-asr/kaldi/pull/2722#pullrequestreview-157516260

xiaohui-zhang commented on egs/tunisian_msa/s5/local/qcri_buckwalter2utf8.sh:
```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+cut -d " " -f 1 qcri.txt > qcri_words_buckwalter.txt
+cut -d " " -f 2 qcri.txt > qcri_prons.txt
```
Actually the option  -f 2 should be -f 2-. I realized this after I saw totally wrong decoding results...

danpovey commented:
I'm closing this PR because I believe we merged it indirectly via #2725.  If there are other changes you want us to merge, please let us know.

johnjosephmorgan commented:
I removed 2 scripts that are not used and 2 config files. And I added a copyright.
conf/pitch.conf
conf/plp.conf
local/buckwalter2utf8.pl
local/qcri_buckwalter2utf8.pl
John


> On Sep 30, 2018, at 2:48 PM, Daniel Povey <notifications@github.com> wrote:
> 
> I'm closing this PR because I believe we merged it indirectly via #2725 <https://github.com/kaldi-asr/kaldi/pull/2725>. If there are other changes you want us to merge, please let us know.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub <https://github.com/kaldi-asr/kaldi/pull/2722#issuecomment-425742314>, or mute the thread <https://github.com/notifications/unsubscribe-auth/ABK_gOfxFNhOYOYXKrEObm3ltCrzm-32ks5ugRH5gaJpZM4Wwdhl>.
>

danpovey commented:
@xiaohui-zhang would you mind making a PR with the latest changes, if appropriate?  No hurry.

xiaohui-zhang commented:
sure!

On Sun, Sep 30, 2018 at 8:31 PM Daniel Povey <notifications@github.com>
wrote:

> @xiaohui-zhang <https://github.com/xiaohui-zhang> would you mind making a
> PR with the latest changes, if appropriate? No hurry.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/kaldi-asr/kaldi/pull/2722#issuecomment-425763952>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ANiEEa2y0aKL2fkc0ml6gY1MKYNSrvrHks5ugWJHgaJpZM4Wwdhl>
> .
>


-- 
Xiaohui
